src/AnalyzeCli.cpp

```diff
@@ -329,10 +331,12 @@ int startAnalyze(const argparse::ArgumentParser& program)
             }
             return 1;
         }
+
+        platform->handleRegisterDefinitions(frontend.globals, types::parseDefinitionsFileMetadata(*definitionsContents));
     }
 
-    types::registerInstanceTypes(frontend, frontend.globals, frontend.globals.globalTypes, fileResolver,
-        !program.is_used("--no-strict-dm-types") && client.configuration.diagnostics.strictDatamodelTypes);
+    platform->handleSourcemapUpdate(
+        frontend, frontend.globals, fileResolver, !program.is_used("--no-strict-dm-types") && client.configuration.diagnostics.strictDatamodelTypes);
```

Comment:
Is working with `RobloxPlatform::instanceTypes` rather than `globals.globalTypes` okay? To be honest, I don't know the purpose of `TypeArena` enough to know what this changes.